src/lib/cache.ts

```diff
@@ -0,0 +1,24 @@
+import NodeCache from 'node-cache';
+
+export const cacheAsync = <T>(
+    fn: () => Promise<T>,
+    ttl: number,
+): [() => void, () => Promise<T>] => {
+    const myCache = new NodeCache();
+    const key = 'res';
```

Comment:
Does the `key` need to be something the caller passes in? Will the default content and test calls overwrite each other since they're using the same key?

Reply:
Yeah I also got the impression both calls would use the same key and overwrite each other

Reply:
We are essentially building a new cache each time `cacheAsync` is called, unique for that call.

This is pretty heavyweight but the assumption is that we are going to cache very few things like this. If we want lots of cached values we'd want to refactor to share the cache between them somehow.

Reply:
(That was a long-winded answer; the short one is that the cache key is pretty irrelevant here because we only ever store one thing in each cache.)

Reply:
Ohh I see, thanks for the explanation. And that works because we're returning the `retFn`. Totally missed that on first read. I wonder if there's a way to make that clearer. Perhaps inlining the key, just because I was drawn to that variable and then surprised it wasn't dynamic. I wonder if renaming to `buildAsyncCache` or something might be helpful?

Reply:
Yeah, I've added a comment to capture the underlying assumption here (only caching a few things).